libs/core/algorithms/include/hpx/parallel/algorithms/rotate.hpp

```diff
@@ -211,30 +214,67 @@ namespace hpx { namespace parallel { inline namespace v1 {
             ExPolicy policy, FwdIter first, FwdIter new_first, Sent last)
         {
             using non_seq = std::false_type;
+            // further inprovements:
+            //add core partition facility
+            //remove unneeded asynchronous code
+
+            std::ptrdiff_t size_left = std::distance(first, new_first);
+            std::ptrdiff_t size_right = std::distance(new_first, last);
+
+            /*
+            add core partition facility:
+                need to deal with 4 cases:
+                cores >=1
+                size_right == 0
+                cores_left>=1 & cores_right >=1
+                cores_left+cores_right = cores;
+            */
+
+            // get number of cores currently used
+            std::size_t cores = execution::processing_units_count(
+                policy.parameters(), policy.executor());
+
+            std::size_t cores_left = 1;
+            if (size_right > 0)
+            {
+                double partition_size_ratio =
+                    double(size_left) / (size_left + size_right);
+                // avoid cores_left =0 after integer rounding
+                cores_left = std::max(
+                    std::size_t(1), std::size_t(partition_size_ratio * cores));
+            }
+            // if size_right==0&cores==1,cores_right =0, should be at least 1.
+            std::size_t cores_right =
+                std::max(std::size_t(1), cores - cores_left);
 
-            auto p = hpx::execution::parallel_task_policy()
-                         .on(policy.executor())
-                         .with(policy.parameters());
+            auto p = policy(hpx::execution::task);
```

Comment:
error report:
/Users/runner/work/hpx/hpx/libs/core/algorithms/include/hpx/parallel/algorithms/rotate.hpp:250:22: error: no matching function for call to object of type 'hpx::execution::parallel_unsequenced_policy'
            auto p = policy(hpx::execution::task);

Reply:
I think this is because of https://github.com/STEllAR-GROUP/hpx/issues/5492. 

At a minimum, we need something like https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/executors/include/hpx/executors/execution_policy.hpp#L1324-L1327,` but just for the `task_policy_tag`. Most likely here as well, BTW:  https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/executors/include/hpx/executors/execution_policy.hpp#L1396-L1399

Reply:
> I think this is because of #5492.
> 
> At a minimum, we need something like https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/executors/include/hpx/executors/execution_policy.hpp#L1324-L1327, but just for the `task_policy_tag`. Most likely here as well, BTW: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/executors/include/hpx/executors/execution_policy.hpp#L1396-L1399

Sorry for the late response, I got 2 drafts [draft_1](https://github.com/hcq9102/hpx/pull/1/files) & [draft_2](https://github.com/hcq9102/hpx/pull/2/files) (ignore the notes) for this issue, but I am not sure which one makes more sense. THANKS.